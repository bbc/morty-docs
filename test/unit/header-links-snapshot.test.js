const React = require('react')
const Header = require('../../src/page-renderers/Components/Header')
const renderer = require('react-test-renderer')

describe('The header is rendered with the correct breadcrumb links ', () => {
  it('When a basepath is passed through', () => {
    const component = renderer.create(
      <Header relPath={'folder/nestedFolder/someFile.html'} basePath={'morty-docs/some-repo'} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('When a basepath is not passed through', () => {
    const component = renderer.create(
      <Header relPath={'folder/nestedFolder/someFile.html'} basePath={''} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
