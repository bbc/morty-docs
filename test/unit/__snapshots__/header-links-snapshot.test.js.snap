// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header is rendered with the correct breadcrumb links  When a basepath is not passed through 1`] = `
<div
  className="container"
  style={
    Object {
      "backgroundColor": "#000",
      "border": "none",
      "borderRadius": "0",
      "height": "60px",
      "marginBottom": "0",
      "padding": "1.142rem",
      "width": "100%",
    }
  }
>
  <nav
    style={Object {}}
  >
    <ol
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "listStyle": "none",
          "padding": "0",
        }
      }
    >
      <li>
        <span
          aria-hidden={true}
          style={
            Object {
              "color": "#ebebeb",
              "margin": "0 0.28rem",
            }
          }
        >
          /
        </span>
        <a
          href="/folder"
          style={
            Object {
              "color": "lightblue",
              "fontSize": "2rem",
              "textAlign": "left",
            }
          }
        >
          folder
        </a>
      </li>
      <li>
        <span
          aria-hidden={true}
          style={
            Object {
              "color": "#ebebeb",
              "margin": "0 0.28rem",
            }
          }
        >
          /
        </span>
        <a
          href="/folder/nestedFolder"
          style={
            Object {
              "color": "lightblue",
              "fontSize": "2rem",
              "textAlign": "left",
            }
          }
        >
          nestedFolder
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`The header is rendered with the correct breadcrumb links  When a basepath is passed through 1`] = `
<div
  className="container"
  style={
    Object {
      "backgroundColor": "#000",
      "border": "none",
      "borderRadius": "0",
      "height": "60px",
      "marginBottom": "0",
      "padding": "1.142rem",
      "width": "100%",
    }
  }
>
  <nav
    style={Object {}}
  >
    <ol
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "listStyle": "none",
          "padding": "0",
        }
      }
    >
      <li>
        <span
          aria-hidden={true}
          style={
            Object {
              "color": "#ebebeb",
              "margin": "0 0.28rem",
            }
          }
        >
          /
        </span>
        <a
          href="/morty-docs"
          style={
            Object {
              "color": "lightblue",
              "fontSize": "2rem",
              "textAlign": "left",
            }
          }
        >
          morty-docs
        </a>
      </li>
      <li>
        <span
          aria-hidden={true}
          style={
            Object {
              "color": "#ebebeb",
              "margin": "0 0.28rem",
            }
          }
        >
          /
        </span>
        <a
          href="/morty-docs/some-repo"
          style={
            Object {
              "color": "lightblue",
              "fontSize": "2rem",
              "textAlign": "left",
            }
          }
        >
          some-repo
        </a>
      </li>
      <li>
        <span
          aria-hidden={true}
          style={
            Object {
              "color": "#ebebeb",
              "margin": "0 0.28rem",
            }
          }
        >
          /
        </span>
        <a
          href="/morty-docs/some-repo/folder"
          style={
            Object {
              "color": "lightblue",
              "fontSize": "2rem",
              "textAlign": "left",
            }
          }
        >
          folder
        </a>
      </li>
      <li>
        <span
          aria-hidden={true}
          style={
            Object {
              "color": "#ebebeb",
              "margin": "0 0.28rem",
            }
          }
        >
          /
        </span>
        <a
          href="/morty-docs/some-repo/folder/nestedFolder"
          style={
            Object {
              "color": "lightblue",
              "fontSize": "2rem",
              "textAlign": "left",
            }
          }
        >
          nestedFolder
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
